examples/platform/linux/AppMain.cpp

```diff
@@ -251,7 +252,9 @@ CHIP_ERROR InitCommissioner()
     params.controllerICAC   = icacSpan;
     params.controllerNOC    = nocSpan;
 
-    ReturnErrorOnFailure(gCommissioner.Init(params));
+    auto & factory = chip::Controller::DeviceControllerFactory::GetInstance();
+    ReturnErrorOnFailure(factory.Init(factoryParams));
+    ReturnErrorOnFailure(factory.SetupCommissioner(params, gCommissioner));
```

Comment:
Here is a possible conflict with the proposed design.

It makes some sense for a commissioner to be a singleton.  For instance, when a commissioner advertises itself with the `_matterd._udp` service type, it's not as though clients of the commissioning service can choose the fabric to which they wish to be commissioned.  There is no authenticated record from which they could make this selection.

And so, we do not expose any fabric-specific information.  Instead, clients choose a commissioner based upon other criteria and the selected commissioner determines itself which fabric the client should be commissioned to.  Thus we probably don't expect a single node to advertise multiple UDC services.  Or at the very least, each advertised UDC service should be understood to be potentially multi-fabric.

But this design can't support that.  We would need each UDC service to map to a single commissioner instance, but yet to we need multiple commissioner instances to support multi-fabric commissioning.  This is only therefore a complete multi-fabric solution for non-commissioning controllers.

Reply:
> This sounds like the same scenario as having multiple commissioners(different physical devices) advertising themselves on the network. The client needs to still be able to make some sort of selection.

To be absolutely clear, it's not the same.  Clients can make a selection, but it's not a fabric selection.  The commissioner, or more specifically a UDC service instance, must make the fabric selection, which therefore implies that a UDC service instance needs to somehow support multi-fabric.

> So this makes sense and I think we can achieve it with a single UDC service that isn't fabric specific. Something that the factory can own. Since UDC requires some user input anyway, a consumer of our SDK will have to provide a UserConfirmationProvider to the factory on init.

Yes, that would work and does get to the finer point that it is actually the UDC service, not the commissioner per se, that needs to be a single instance that can somehow map to multiple fabrics.

Followup is absolutely fine.  I only wanted us to consider this (admittedly obscure) use case and have some plan to potentially address it.